packages/ui/build/common-config.ts

```diff
@@ -52,8 +62,8 @@ const libBuildOptions = (format) => ({
   },
 })
 
-export default function createViteConfig (format) {
-  const isEsm = ['esm', 'esm-node'].includes(format)
+export default function createViteConfig (format: BuildFormat): UserConfig {
+  const isEsm = ['es', 'esm-node'].includes(format)
   const isMjs = format === 'esm-node'
 
   const config = {
```

Comment:
Sorry, that won't work.

UserConfig has pretty much all fields optional, which doesn't let me `config.field.push(stuff)` as TS would assume field could be `undefined`. If don't declare config on initialization - TS allows me to narrow down declaration on particular fields.

Reply:
You can use `config.build!.push`. UserConfig here makes useful suggestion. 

It can be replaced with `defineViteConfig<T extends UserConfig>(config: T): T` if you want to prevent `!`.